src/routers/alpha-router/alpha-router.ts

```diff
@@ -606,7 +610,8 @@ export class AlphaRouter
     quoteCurrency: Currency,
     tradeType: TradeType,
     swapConfig?: SwapConfig,
-    partialRoutingConfig: Partial<AlphaRouterConfig> = {}
+    partialRoutingConfig: Partial<AlphaRouterConfig> = {},
+    swapAndAddParameters?: SwapAndAddParameters,
```

Comment:
Hmm I think it would be cleaner to leave route as is, and make routeToRatio responsible for generating the swapAndAdd calldata. Feels like its a cleaner separation of concerns to have IRouter/route purely responsible for swap calldata, and ISwapToRatio/routeToRatio responsible for swap+liquidity calldata.

Reply:
woops just seeing this. addressing.